app/javascript/components/SearchPanel.js

```diff
@@ -1,44 +1,43 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 
+import KeywordSearch from './KeywordSearch';
 import FacetControl from './FacetControl';
 import MoreFacetsButton from './MoreFacetsButton';
 import DownloadButton from './DownloadButton';
-import KeywordSearch from './KeywordSearch';
 
-// Only for development!  We'll fetch data once API endpoints are available.
-import {facetsResponseMock, searchFiltersResponseMock} from './FacetsMockData';
-const facets = facetsResponseMock;
+import { fetchFacets } from './../lib/scp-api';
 
 const defaultFacetIDs = ['disease', 'organ', 'species', 'cell_type'];
 const moreFacetIDs = ['sex', 'race', 'library_preparation_protocol', 'organism_age'];
 
-const defaultFacets = facets.filter(facet => defaultFacetIDs.includes(facet.id));
-const moreFacets = facets.filter(facet => moreFacetIDs.includes(facet.id));
-
-window.searchFiltersResponse = searchFiltersResponseMock;
-
-// const searchStyle= {
-//   'font-size':'22px',
-//   color: '#333F52'
-
-// }
-// const searchPanelStyle = {
-//   borderRadius: '25px',
-//   background: 'white'
-// };
 /**
- * Component for SCP advanced search UI
+ * Component for SCP faceted search UI
  *
  * This is the entry point into React code from the traditional JS code
  * See related integration at /app/javascript/packs/application.js
  */
-function SearchPanel() {
-  // Note:  Enventually this fuction will have State and will turn into a class component. There's room for this to become 
-  // a higher order Component (HOC). This Search component is specific to the "Studies"
-  // tab when it should be able to support the 'home' Seach Panel, Studies, Genes and Cells search panels.
+export default function SearchPanel() {
+  // Note: This might become  a Higher-Order Component (HOC).
+  // This search component is currently specific to the "Studies" tab, but
+  // could possibly also enable search for "Genes" and "Cells" tabs.
+
+  const [defaultFacets, setDefaultFacets] = useState([]);
+  const [moreFacets, setMoreFacets] = useState([]);
+
+  useEffect(() => {
```

Comment:
is there a reason this function has to be declared inside the useEffect?  I could see having it defined outside so that useEffect is essentially a plain list of 'do these things after render' .   Then again, if it's important to not have the data ever fetched outside of this hook, the scoping makes sense

Reply:
Interesting point!  This code will likely evolve quickly as the initialization logic for facets is refined.  I've noted this as something to consider in SCP-2109.